server/vcr-server/api/v3/serializers/search.py

```diff
@@ -1,135 +1,156 @@
 import logging
 
-from rest_framework.serializers import CharField, SerializerMethodField
-from drf_haystack.serializers import (
-    HaystackFacetSerializer,
-    HaystackSerializerMixin,
-    HaystackSerializer,
-)
+from rest_framework.serializers import SerializerMethodField
+from drf_haystack.serializers import HaystackSerializer
 
 from ..indexes.Address import AddressIndex
 from ..indexes.Name import NameIndex
 from ..indexes.Topic import TopicIndex
 
-from api.v2.models.Address import Address
-from api.v2.models.Name import Name
-
-from api.v2.serializers.rest import NameSerializer, AddressSerializer, TopicSerializer
+from api.v2.serializers.rest import (
+    AddressSerializer,
+    NameSerializer,
+    TopicSerializer
+)
 
 
 logger = logging.getLogger(__name__)
 
 
-class NameAutocompleteSerializer(HaystackSerializer):
+class AriesSearchSerializer(HaystackSerializer):
 
     type = SerializerMethodField()
+    sub_type = SerializerMethodField()
     value = SerializerMethodField()
-    score = SerializerMethodField()
     topic_source_id = SerializerMethodField()
+    # DEPRECATED
     topic_type = SerializerMethodField()
+    credential_type = SerializerMethodField()
     credential_id = SerializerMethodField()
 
     @staticmethod
     def get_type(obj):
-        return "name"
+        return None
 
     @staticmethod
-    def get_value(obj):
-        return obj.name_text
+    def get_sub_type(obj):
+        return None
 
     @staticmethod
-    def get_score(obj):
-        return obj.score
+    def get_value(obj):
+        return None
```

Comment:
Would it be more appropriate to have the body of these methods set to `pass` rather than returning none?
The way this is setup looks like an interface/abstract class, but so that could be appropriate, but I may be reading this wrong perhaps.

Reply:
I think that may be a better representation. Assuming a user would extend this class it would be expected that they implement these methods. An abstract class would 'enforce' an implementation.

Reply:
Yeah, that is what I was thinking (also my comment seems to have been garbled up - not sure what I typed there now that I re-read it). I was thinking pass would require an implementation to be provided, while returning `None` would compile and potentially cause issues, but I am not sure what the right way to do this in python is.